catalog/app/components/JsonEditor/Input.js

```diff
@@ -77,6 +93,19 @@ export default function Input({
   const [valueStr, setValueStr] = React.useState(() =>
     getNormalizedValueStr(originalValue, data.valueSchema),
   )
+  const [firstTime, setFirstTime] = React.useState(true)
```

Comment:
i believe `useState` is not the appropriate hook for this, we should rather use `useRef`

Reply:
or you could even try not using this variable at all, just call `useLayoutEffect` with an empty dependency array -- this way it will only be called on mount (similar to `componentDidMount`).
also, why are you using `useLayoutEffect` instead of regular `useEffect` here?

Reply:
> why are you using `useLayoutEffect`

I want to be sure that all DOM jobs are completed. I can add a TODO comment here, to replace it with a regular `useEffect`, but I think I should invest more time in testing it

Reply:
Sure, `useEffect(func, [])` - is a right choice, thanks